test/Tests/src/Data/Text_Spec.enso

```diff
@@ -1402,141 +1402,93 @@ spec =
     Test.group "Text.replace" <|
         Test.specify "should work as in examples" <|
             'aaa'.replace 'aa' 'b' . should_equal 'ba'
-            "Hello World!".replace "[lo]" "#" matcher=Regex_Matcher.Value . should_equal "He### W#r#d!"
-            "Hello World!".replace "l" "#" mode=Matching_Mode.First . should_equal "He#lo World!"
-            '"abc" foo "bar" baz'.replace '"(.*?)"' '($1)' matcher=Regex_Matcher.Value . should_equal '(abc) foo (bar) baz'
-            'ß'.replace 'S' 'A' matcher=Text_Matcher.Case_Insensitive . should_equal 'AA'
-            'aﬃb'.replace 'i' 'X' matcher=Text_Matcher.Case_Insensitive . should_equal 'aXb'
+            "Hello World!".replace "[lo]" "#" . should_equal "He### W#r#d!"
+            "Hello World!".replace "l" "#" only_first=True . should_equal "He#lo World!"
```

Comment:
Most of the tests are using:
- constant input string
- constant regex string
- fixed parameters

I assume that's not the use-case we want to optimize on. What we want is:
- constant regexp string
- fixed parameters
- **variable input** strings send in many times

Right? I assume the proposal behind `Replacer` is to write such code _differently_. I haven't found an example however.

Reply:
Are you talking about the goal of not compiling regexes more than once? I could add a test case that applies a regex to multiple strings, but as far as I know, a test here cannot detect whether or not the regex is being compiled more than once, so it wouldn't be any better.

Reply:
A test or sample code showing how to process many data with a single regex replace is what I am looking for. 

> test here cannot detect whether or not the regex is being compiled more than once

Writing such a test is certainly not easy.